src/libsystemd-network/dhcp-static-lease.c

```diff
@@ -0,0 +1,102 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "alloc-util.h"
+#include "dhcp-internal.h"
+#include "dhcp-server-internal.h"
+#include "memory-util.h"
+#include "strv.h"
+#include "utf8.h"
+
+static sd_dhcp_static_lease *sd_dhcp_static_lease_free(sd_dhcp_static_lease *i) {
+        if (!i)
+                return NULL;
+
+        free(i->client_id.data);
+        return mfree(i);
+}
+
+static int dhcp_static_lease_new(
+                DHCPClientId *client_id,
+                const struct in_addr *address,
+                sd_dhcp_static_lease **ret) {
+
+        assert_return(ret, -EINVAL);
+        assert_return(address, -EINVAL);
+        assert_return(client_id, -EINVAL);
+
+        sd_dhcp_static_lease *p = new(sd_dhcp_static_lease, 1);
+        if (!p)
+                return -ENOMEM;
+
+        *p = (sd_dhcp_static_lease) {
+                .n_ref = 1,
+                .client_id = *client_id,
+                .address = le32toh(address->s_addr)
+        };
+
+        *ret = TAKE_PTR(p);
+        return 0;
+}
+
+static int dhcp_static_lease_new_default(sd_dhcp_static_lease **ret) {
+        struct in_addr address = {};
+        DHCPClientId client_id = {};
+        return dhcp_static_lease_new(&client_id, &address, ret);
+}
+
+int sd_dhcp_static_lease_set_client_id_by_mac(
+                sd_dhcp_static_lease *lease,
+                const uint8_t *mac_addr,
+                size_t mac_addr_size) {
+
+        assert_return(lease, -EINVAL);
+        assert_return(mac_addr, -EINVAL);
+        assert_return(mac_addr_size, -EINVAL);
+
+        void *data = malloc(mac_addr_size + 1);
```

Comment:
Same here. Please move `void *data;` before the assertions.

Also, I think it should be `uint8_t *` instead of `void *`. See the comment below.